### PR TITLE
Fix sophia-oracle-introduction.md contract

### PR DESCRIPTION
**Pull request by eltNEG**
**Description**

It fixed the outdated contract in sophia-oracle-introduction.md.

**Tasks completed**

- [x] Fix the Sophia contract in sophia-vote-contract.md
- [x] Correct md linting errors and warning

**Notes**

None

### CODE REVIEW

File: sophia-oracle-introduction.md [hunk 1]
```diff
@@ -26,128 +28,157 @@ contract Greeter =
...
 Now that we are able to query our oracle, we should also add logic to respond to those queries. We will have to work with a recursive function that iterates through all of the queries and responds to them. To make it a bit more complicated, we will make it so that the oracle responds only to certain greets, let's say "Hello", "Hi" and "Greetings".
 
 Let's begin with the public function that will start the recursive calls:
-```
-public function respond_to_greets() =
-    respond_to_greets'(state.greets)
+
+```sophia
+stateful entrypoint respond_to_greets() =
+  respond_to_greets'(state.greets)
```
Comment (hanssv) [inline sophia-oracle-introduction.md]:
```suggestion
  [ respond_to_greet(caller_addr, query) | (caller_addr, query) <- state.greets) ]
```

Comment (hanssv) [reply sophia-oracle-introduction.md]: Will simplify the code and you can skip the explanation about not being able to have to functions with the same name below.

File: sophia-oracle-introduction.md [hunk 2]
```diff
@@ -26,128 +28,157 @@ contract Greeter =
...
+  payable stateful entrypoint greet_oracle(message : string) : bool =
+    let val = Call.value
+    if(val != 10)
+      false
+    else
+      let query : oracle_query(string, string) =
+        Oracle.query(state.greeter_oracle, message, 10, RelativeTTL(50), RelativeTTL(50))
+      let caller_address : string = Address.to_str(Call.caller)
+      let new_greets = (caller_address, query) :: state.greets
+      put(state{greets = new_greets})
```
Comment (hanssv) [inline sophia-oracle-introduction.md]: I liked `put(state{greets @ g = (caller_address, query) :: g})` better ;-)

File: sophia-oracle-introduction.md (hunk 1 above)
Comment (eltNEG) [reply sophia-oracle-introduction.md]: Thanks for the feedback @hanssv

` [ respond_to_greet(caller_addr, query) | (caller_addr, query) <- state.greets) ]`

a similar thing [respond_to_greet(c,q) for c,q in state[greets]] is called list cmprehension in python. What is this called in sophia?

Comment (thepiwo) [reply sophia-oracle-introduction.md]: The concept in general is called list comprehension. https://github.com/aeternity/protocol/blob/master/contracts/sophia.md#lists

**Pull request merged**